scripts/globals/garrison.lua

```diff
@@ -0,0 +1,133 @@
+-----------------------------------
+-- Garrison
+-----------------------------------
+require("scripts/globals/common")
+require("scripts/globals/garrison_data")
+require("scripts/globals/npc_util")
+require("scripts/globals/settings")
+require("scripts/globals/status")
+require("scripts/globals/utils")
+require("scripts/globals/zone")
+-----------------------------------
+xi = xi or {}
+xi.garrison = xi.garrison or {}
+xi.garrison.lookup = xi.garrison.lookup or {}
+
+-----------------------------------
+-- Helpers
+-----------------------------------
+local spawnMobs = function(mobList)
+    for _, mobId in ipairs(mobList) do
+        SpawnMob(mobId)
+    end
+end
+
+local despawnMobs = function(mobList)
+    for _, mobId in ipairs(mobList) do
+        DespawnMob(mobId)
+    end
+end
+
+-----------------------------------
+-- Main Functions
+-----------------------------------
+xi.garrison.tick = function(npc)
+    print("Garrison Tick")
+
+    -- Extract data from lookup
+    local zoneId = npc:getZoneID()
+    local garrisonData = xi.garrison.lookup[zoneId]
+
+    -- Check all registered Players are dead or invalid -> lose
+    local setupTimer = false
+
+    -- Check all NPCs are dead -> lose
+
+    -- Check all Mobs are dead -> win
+
+    -- Check Mobs in current wave are dead -> next wave
+    local nextWave = 1
+    xi.garrison.spawnWave(npc, nextWave)
+
+    -- Make NPCs attack Mobs
+
+    -- Make Mobs attack NPCs
+
+    -- Tick again?
+    if setupTimer then
+        npc:timer(2400, function(npcArg)
+            xi.garrison.tick(npcArg, false)
+        end)
+    end
+end
+
+xi.garrison.spawnNPCs = function(npc)
+    local zoneId = npc:getZoneID()
+    local garrisonZoneData = xi.garrison.data[zoneId]
+
+    -- Build npcList
+    local npcList = {}
+    for _, npcId in ipairs(garrisonZoneData.npcs) do
+        table.insert(npcList, npcId)
+    end
+
+    spawnMobs(npcList)
+
+    -- Apply confrontation effect
```

Comment:
Unless you got a cap showing me its confrontation effect (which means SE updated it) it is not confrontation effect :P

I did tons of these before that effect even existed in the client, they def did not use that effect.

Reply:
Eu Sou tudo de bom